### PR TITLE
4 domaca zadaca

### PR DESCRIPTION
**Pull request by amarkotic**
Pitanje: Prilikom kreiranja SearchViewControllera u AppRouteru i dodavanja istog u TabBarController kao sredisnji controller nailazim na problem. Da bih iz SearchViewControllera mogao pokrenuti kviz, moram ga kreirati kao UINavigationController kako bi pushanje kviza na vrh stacka bilo moguće. Iz nekog čudnog razloga kada to učinim, SearchViewController mi postane izrazito neresponzivan pri pokretanju na fizičkom uređaju, dok se čudno ponaša na simulatoru(ne želi pokrenuti kviz, ali je responzivan). A čim SearchViewController predam TabControlleru kao običan UIViewController, a ne UINavigationController, on postaje responzivan na fizičkom uređaju samo što sada naravno nije moguće pokrenuti kviz jer je push nemoguć.

Dakle za ispitavnja što se događa treba pogledati u AppRouter pri kreiranju SearchViewControllera i dodavanja istog u TabBar.

Nisam uspio sam ustanoviti zašto se to događa pa bih cijenio bilo kakav komentar i pomoć sa tim problemom.

### CODE REVIEW

File: QuizApp/Presenter/QuizzesViewPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        self.nService = networking
+    }
+    func setquizzesViewDelegate(quizzesViewDelegate : QuizzesViewController?){
+        self.quizzesViewDelegate = quizzesViewDelegate
+    }
+    
+    
+    
+    
+    func fetchQuizzes(with internetConnection: Bool){
```
Comment (istaudacher) [inline QuizApp/Presenter/QuizzesViewPresenter.swift:40]: This logic of fetching data from API and saving it to CoreData should be in Repository. The presentation layer should not be aware of this data handling - that is the responsibility of the Data layer.

File: QuizApp/Presenter/Coordinators/AppRouter.swift [hunk 2]
```diff
@@ -43,16 +43,20 @@ class AppRouter: AppRouterProtocol{
         let qvc = QuizzesViewController(router: self)
         let navQvc = UINavigationController(rootViewController: qvc)
         let svc = SettingsViewController(router: self)
-        
-        //tabbar[QuizzesViewControler, SettingsViewController]
+        let searchVc = SearchViewController(router: self)
+        let navSearchVc = UINavigationController(rootViewController: searchVc)
```
Comment (istaudacher) [inline QuizApp/Presenter/Coordinators/AppRouter.swift:47]: You do not need to wrap each VC into UINavigationController for this type of application design.

File: QuizApp/View/SearchView/SearchViewController+Design.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+//  Created by Antonio Markotic on 01.06.2021..
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+
+extension SearchViewController{
+    func buildViews(){
```
Comment (istaudacher) [inline QuizApp/View/SearchView/SearchViewController+Design.swift:14]: It should be loadElements and then setConstraints.

File: QuizApp/View/SearchView/SearchViewController+Design.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        searchStack.snp.makeConstraints { make in
+            make.top.equalTo(view).offset(100)
+            make.centerX.equalTo(view)
+            make.width.equalTo(view.frame.width*0.90)
+           
+        }
+       
+        
+        //table view
+        view.addSubview(tableView)
```
Comment (istaudacher) [inline QuizApp/View/SearchView/SearchViewController+Design.swift:42]: This should be in loadElements.

File: QuizApp/View/SearchView/SearchViewController+Design.swift [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            make.width.equalTo(view.frame.width * 0.2)
+            make.height.equalTo(40)
+        }
+        
+        inputTextField.snp.makeConstraints { make in
+            make.height.equalTo(40)
+        }
+        
+        //search stack
+        view.addSubview(searchStack)
```
Comment (istaudacher) [inline QuizApp/View/SearchView/SearchViewController+Design.swift:32]: This should be in loadElements.

File: QuizApp/View/SearchView/SearchViewController.swift [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    
+}
+
+
+
+
+
+extension SearchViewController: UITableViewDelegate{
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```
Comment (istaudacher) [inline QuizApp/View/SearchView/SearchViewController.swift:114]: This func is never called on quiz click so starting quiz from Search does not work.

Comment (amarkotic) [reply QuizApp/View/SearchView/SearchViewController.swift:114]: I understand how to fix mistakes in other comments, but this one bothers me still.

I wrote it in the Pull request Readme : "...dok se čudno ponaša na simulatoru(ne želi pokrenuti kviz, ali je responzivan)."

If you run the app on a physical device, SearchVC recognizes a click and opens the quiz, but not from the first try(that's why I wrote it's unresponsive on physical device). I have to fast click few times and than the quiz appears from the searchVC.

Also, is saving data to CD and fetching from CD okay? No comments on that but would like a feedback.

**Pull request merged**